addons/l10n_ch/models/res_bank.py

```diff
@@ -34,6 +34,25 @@ class ResPartnerBank(models.Model):
     _inherit = 'res.partner.bank'
 
     l10n_ch_postal = fields.Char(string='ISR reference', help='The ISR number of the company within the bank')
+    acc_type = fields.Selection(
+        selection=lambda x: x.env['res.partner.bank'].get_supported_account_types(),
+        compute='_compute_acc_type',
+        store=True,
+        string='Type',
+        help='Bank account type: Normal or IBAN. Inferred from the bank account number.'
+    )
+
+    @api.model_cr_context
+    def _auto_init(self):
+        res = super()._auto_init()
+        tools.drop_constraint(self._cr, self._table, 'res_partner_bank_unique_number')
```

Comment:
Are you kidding me?

Reply:
@jco-odoo there is no way we can achieve our goal(`unique account numbers per company if account type(acc_type) is IBAN and allow duplicate account numbers if acc_type is postal(Normal)`) by keeping this constraint as postgres does not support partial constraints, so IMO this is the only way we can achieve our goal. Do you think there could be another way ?
Also, I see we such created unique index at few places like <a href="https://github.com/odoo/odoo/blob/master/odoo/addons/base/models/ir_translation.py#L256">ir_translation.py</a>

Reply:
In Switzerland, either you pay using the IBAN or you pay by structured payment ref (BVR).

In this 2nd case, you need the postal account of the bank agency (instead of the IBAN).
This postal acc. is NOT unique since all the customer or the bank agency share this account number.
This 2nd payment mode is by far the most used in Switzerland so it is not an Except at all.

So we have 2 options here in my view : either you remove the unicity constraint or you create a new field to store this postal acc.

Reply:
@fclementic2c There is already a field like this, so if you use IBAN accounts, the field ISR reference is automatically deduced.  In the case of postal account, the fields are equal by the onchange, but in principle you could put something else in the account number field then.  

How does the bank determine the customer in these cases?  (name of the partner, ...)

Reply:
@jco-odoo 
Ok good. I see the field ISR Reference you are talking about but there are 2 fields with this name : on on the res.bank and one on the res.partner.bank.

There is a confusion here because:
* the one on the res.parner.bank should be to store the supplier postal account number (in case you want to pay a supplier)
* the one on the res.bank should be to store the adherent number (in this case you want your own company being able generate ISR invoices. This adherent number is part of the ISR ref.

Currently (I did the test here: http://568775-12-0-e33fe9.runbot18.odoo.com/web), you must put the same reference on both. if one is missing you get an message when you click on print ISR.

Also, there is no way to compute the ISR reference on the res.partner.bank because it is not based on the IBAN at all... The ISR reference field should not be computed (in switzerland at least).

Is it more clear for you ?

I am fully available on this topic for you if you need it. This is top priority for me.

Reply:
@fclementic2c @jcr-odoo We are talking about the res.partner.bank here.  The trouble is the acc_number is duplicated and when you change the acc_number on postal it changes the isr reference to the acc_number.  As the acc_number is unique and constraints are technically difficult to change this gives us problems.  

We can undo that uniqueness but that requires changing code elsewhere.  (it takes time to do it properly)

I don't know if we can just remove or flexibilize the onchange a bit, so you can enter a unique name in the acc_number yourself, while not changing the postal.  But not immediately sure that would do the trick.

Reply:
@jco-odoo I am back from holidays. What about you guys ? Can we go forward on this please ?

Reply:
@fclementic2c No, read my comments.  We did some alternative branches, but no idea if they would do the trick.  (no time to test)

Reply:
@fclementic2c Normally you can already do it, because you can put something unique in acc_number and the real postal in the ISR reference.  It just that it is probably weird with the onchanges.  And if we don't have to drop the constraint, this can maybe go in stable.  If you have time next week...

Reply:
Ok I will check this asap thxs. (but what you say seems to be the good solution to me)